File: frontend/src/types/assignment.ts
export interface AssignmentDTO {
  id: string;
  course_id: string;
  module_id: string;
  title: string;
  description: string;
  max_points: number;
  due_date: string | null;
  allow_late_submission: boolean;
  position: number;
  created_at: string;
  updated_at: string;
}

export interface SubmissionDTO {
  id: string;
  assignment_id: string;
  student_id: string;
  text_content: string;
  link_url: string;
  file_path: string;
  score: number | null;
  feedback: string;
  graded_at: string | null;
  submitted_at: string;
}

export interface CreateAssignmentInput {
  title: string;
  description?: string;
  max_points: number;
  due_date?: string;
  allow_late_submission?: boolean;
}

export interface UpdateAssignmentInput {
  title?: string;
  description?: string;
  max_points?: number;
  due_date?: string;
  allow_late_submission?: boolean;
}

File: backend/internal/domain/entities/assignment.go
package entities

import (
	"time"

	"github.com/google/uuid"
)

type Assignment struct {
	ID                  uuid.UUID
	OrgID               uuid.UUID
	CourseID            uuid.UUID
	ModuleID            uuid.UUID
	Title               string
	Description         string
	MaxPoints           int
	DueDate             *time.Time
	AllowLateSubmission bool
	Position            int
	CreatedAt           time.Time
	UpdatedAt           time.Time
}

type AssignmentSubmission struct {
	ID           uuid.UUID
	AssignmentID uuid.UUID
	StudentID    uuid.UUID
	OrgID        uuid.UUID
	TextContent  string
	LinkURL      string
	FilePath     string
	Score        *int
	Feedback     string
	GradedBy     *uuid.UUID
	GradedAt     *time.Time
	SubmittedAt  time.Time
	CreatedAt    time.Time
	UpdatedAt    time.Time
}
